app/models/event.rb

```diff
@@ -0,0 +1,18 @@
+class Event < ApplicationRecord
+	#Allowing URL, end_date, and address2 to be empty 
+	validates :name, presence: true
+	validates :description, presence: true
+	validates :start_date, presence: true
+	validates :address1, presence: true
+	validates :city, presence: true
+	validates :state, presence: true, length: {is: 2} #Dropdown selection on front end?  
+	VALID_ZIP_REGEX= /\A\d{5}-?\d{4}?\z/
+	validates :zip, presence: true, format:{with:VALID_ZIP_REGEX}
+	validates :scholarship_available, inclusion: {in: [true, false]}
+	VALID_URL_REGEX = /\Awww\.+.+\.[a-z]+\z/i 
+	validates :URL, format: {with:VALID_URL_REGEX}
```

Comment:
What do you think about using `URI::Regexp` here? It would make people include `http://` or `https://` but we would have a tried and true regexp. We could also use `URI.parse` to inspect the input a bit.